File: backend/app/models/github.py
from pydantic import BaseModel
from typing import List, Optional

class User(BaseModel):
    login: str

class Repository(BaseModel):
    name: str
    full_name: str

class PullRequest(BaseModel):
    id: int
    number: int
    title: str
    html_url: str
    user: User
    repository: Repository
    hasTests: Optional[bool] = None  # Now optional to support lazy loading

class RequestTestsPayload(BaseModel):
    owner: str
    repo: str
    prNumber: int
    comment: str | None = None  # Optional for the initial request, required for posting 

File: app/types/github.ts
export interface User {
    login: string;
}

export interface Repository {
    name: string;
    full_name: string;
}

// Raw API response interface
export interface RawPullRequestData {
    id: number;
    number: number;
    title: string;
    html_url: string;
    user: User;
    repository: Repository;
    hasTests?: boolean;
}

export interface PullRequest {
    id: number;
    number: number;
    title: string;
    html_url: string;
    url: string;  // For the component's href
    user: User;
    repository: Repository;
    hasTests?: boolean;  // Optional to match backend model
    owner: string;  // Derived from repository.full_name
    repo: string;   // Derived from repository.name
}

export interface RequestTestsPayload {
    owner: string;
    repo: string;
    prNumber: number;
    comment?: string;  // Optional for initial request, required for posting
} 